### DISCUSSION TITLE
What mpd version do you use?

### DISCUSSION THREAD

**Discussion by jcorporation** [jcorporation/myMPD#438] [General]
I am planing to support only MPD 0.21.x or newer to drop some compatibility code in myMPD.

MPD 0.21 has been released 2018/10/31.

**Comment by Emilot** [comment]
v0.22.6

+1 for the "drop" thing...

**Reply by jcorporation** [reply]
Yeah, I am in the process of streamlining the code.

**Comment by parkmino** [comment]
I'm using the lightweight mpd 0.20.23 since I do not like mpd 0.21 for personal reason.
I've never heard of any mpd clients supporting only for mpd 0.21 and later.
Please consider mpd 0.20 support as now.

**Reply by jcorporation** [reply]
Someone must be the first ;) but the decision is not final. I wait for more responses.

Why do you stick with an old MPD version? Newer MPD versions provide many enhancements. 

myMPD has a feature detection to disable features that uses newer MPD functions. More and more features will be available only with newer MPD versions. I do not want to write workarounds for outdated software.

**Reply by parkmino** [reply]
Actually I am not interested in new features, only lightweight and sound matters.
Meson and new features are not attractive to me.
I've recently built the latest mpd git to test Windows wasapi, however I will stick to mpd 0.20 under Linux.
I don't want to reboot just to test myMPD under WSL ;-)

**Reply by Emilot** [reply]
and there is a difference in sound between 0.20 and 0.21?

i am sure there isn't...

**Reply by parkmino** [reply]
I hear the difference, however some say yes, some say no.
It is a controversial area.

**Reply by Emilot** [reply]
but you should agree that if someone knows the absolute truth about that, this is Max who made the MPD.

**Reply by parkmino** [reply]
Unlike me, he says no in this case.
He is not perfect in every way.

**Comment by jcorporation** [comment]
For myMPD v7.0.0 the compatibility code for MPD < 0.20.0 is remove. myMPD prints an error and exits.

**Comment by jcorporation** [comment]
> I've never heard of any mpd clients supporting only for mpd 0.21 and later.

mpc 0.34 requires now mpd 0.21 or above, next ncmpc version also.

- https://github.com/MusicPlayerDaemon/ncmpc/blob/master/NEWS
- https://github.com/MusicPlayerDaemon/mpc/blob/master/NEWS

**Comment by parkmino** [comment]
Hi!

I use mpd 0.20.23 and read the changelog for mpc.
It might be the first mpd client which supports only for mpd 0.21 and later.
It is up to you whether to support the old mpd versions or not.

Are you ready for mympd 8.2.0?
I am about to upload the Korean translation for this.

Thanks

Minho Park


2021년 12월 6일 (월) 오전 4:24, Jürgen Mang ***@***.***>님이 작성:

**Comment by jcorporation** [comment]
Actually I have no pain to support mpd 0.20.x. The 8.x Series of myMPD will therefore support mpd 0.20.x with some disabled features (smart playlists, database view).

v8.2.0 is in the stabilization phase, only bugfixes and small enhancements are made. I hope that v8.2.0 will be released this or next week.

**Comment by GK-774** [comment]
"8.2.0 is in the stabilization phase" - after having read this I tried it today.
First it´s apparently not a seamless upgrade as it used to be in the past, after the upgrade the screen is stuck at "Initializing". Ok, so I removed all entries from /var/lib/mympd/state and it loads, works (and looks nice). Reconfigured from scratch (colors, homeicons, features) and tried on a tablet (ipad) - stuck at "Initializing". Tried another tablet (android) - stuck at "Initializing". Did a reload of the page on the Laptop (Firefox) - stuck at "Initializing". I´m now back to 8.1.6 which works as expected. It´s unfortunate that it will not receive any bugfixes until 8.2.x is ready for daily use.

**Reply by jcorporation** [reply]
I can not reproduce your issue. I work daily with 8.2.0 and it feels pretty stable. We should debug your issue before 8.2.0 gets released. Please open an issue and attach debug info as requested in the template.

Stuck at "Initializing" is mostly an javascript issue. The output of the browser console should point to the problem.

**Comment by GK-774** [comment]
My apologies, a 2nd try on a small dedicated test machine worked ok. (The first try was a machine with selinux enabled and an apache httpd before it. I suspect selinux prevents the new location of the covercache. I will investigate that when I have the time, but it´s not directly myMPD related.) Very nice improvements to the interface!
I had to redo one home icon, a album view with the newest albums descending ("Validation of value for JSON path "$.params.sort" has failed"), no big deal, works after doing it new.

One thing, I use mpd 0.22.11 which is the latest of the 0.22 series and less than 4 months old. Calling that "MPD version is outdated" is a bit exaggerated. The 0.23 mpd series is still receiving a lot of bugfixes (5 releases in 7 weeks) and not everyone is prepared to track that yet. 0.22.11 is certainly not outdated but rather recent and very stable.

**Reply by jcorporation** [reply]
Great that 8.2.0 works for you. The outdated warning is displayed for all mpd versions older than 0.23.5, because:
- there are no bugfixes for the 0.22 branch
- myMPD needs 0.23.5 for some functions: https://github.com/jcorporation/myMPD/blob/devel/docs/references/feature-matrix.md

**Reply by GK-774** [reply]
Well, no bugfixes for the 0.22 branch, I´m not aware of any open bugs for it, either. Anyway, thanks for info on the functions, that´s an argument for using 0.23 from now on, will give it a try.

**Comment by GK-774** [comment]
While it works on my CentOS8 testsystem, on CentOS7 I cannot get it to work. I disabled selinux temporarily but no avail. The Java script errors are:
combined.js:8 ERROR: Locale denot defined
logLog @ combined.js:8
combined.js:8 ERROR: JavaScript error: Uncaught TypeError: Cannot read properties of undefined (reading 'dispose') (http://mpd.hal9000.at:8080/js/combined.js: 8)

**Reply by jcorporation** [reply]
We should discuss this error not in this discussion, that is off topic. I opened #630 for this.

**Comment by vernonjvs** [comment]
> I am planing to support only MPD 0.21.x or newer to drop some compatibility code in myMPD.
> 
> MPD 0.21 has been released 2018/10/31.

I primarily use mpd 0.21.5-3 which is the latest Raspberry PI OS version for the Raspberry Pi Zero W

**Comment by DoDiesis** [comment]
I use version 0.23.5 on my DAP, made using an Intel NUC with a Celeron CPU, running a customized (by me) Archlinux distribution.

**Comment by jcorporation** [comment]
With devel myMPD supports the new priority sorting and filtering of MPD 0.24

**Comment by bonelifer** [comment]
Version 0,23.7 compiled from git source. Needed the newest version(or at least a newer version than the one released with Ubuntu 20.04) to support mpd2chromecast MPD HTTP streaming. On Xubuntu 20.04.4 LTS.
